source/mir/random/flex/internal/area.d

```diff
@@ -0,0 +1,668 @@
+module mir.random.flex.internal.area;
+
+import mir.random.flex.internal.types : Interval;
+import std.traits : ReturnType;
+
+/**
+Determines the hat and squeeze function of an interval.
+Based on Theorem 1 of Botts et al. (2013).
+*/
+void determineSqueezeAndHat(S)(ref Interval!S iv)
+{
+    import mir.utility.linearfun : linearFun;
+    import mir.random.flex.internal.types : determineType, FunType;
+
+    // y (aka x0) is defined to be the maximal point of the boundary points
+    enum sec = `(iv.ltx >= iv.rtx) ?
+                linearFun!S((iv.rtx - iv.ltx) / (iv.rx - iv.lx), iv.lx, iv.ltx) :
+                linearFun!S((iv.rtx - iv.ltx) / (iv.rx - iv.lx), iv.rx, iv.rtx)`;
+
+    enum t_l = "linearFun!S(iv.lt1x, iv.lx, iv.ltx)";
+    enum t_r = "linearFun!S(iv.rt1x, iv.rx, iv.rtx)";
+
+    // could potentially be saved for subsequent calls
+    FunType type = determineType(iv);
+    with(FunType) with(iv)
```

Comment:
I repeat my earlier suggestion that the `with` statements here should be removed.  What they save in typing, they remove in clarity of what's being altered (i.e. there is nothing in the code as written to suggest that `squeeze` and `hat` belong to `iv`).